### ISSUE TITLE
Using expand-region in an Org-mode document makes unrelated sections with code visible

### ISSUE THREAD

**Issue by teodorlu** [magnars/expand-region.el#241]
Hello!

Found some funny behaviour.

# Description

I have this org document.

```org
* Unrelated section without code
This section just has some text in it, and is not shown when expand-region is used.
* Unrelated section with code
... that contains some code:

#+BEGIN_SRC python
  print("Just some code!")
#+END_SRC

This section is expanded when expand-region is used.
* Section where expand-region is used
Let's say we have a superlongword. I want to select superlongword with
expand-region. When I =M-x expand-region= to select superlongword, the unrelated
section 
```

I collapse the first to sections,

```org
* Unrelated section without code...
* Unrelated section with code...
* Section where expand-region is used
Let's say we have a superlongword. I want to select superlongword with
expand-region. When I =M-x expand-region= to select superlongword, the unrelated
section 
```

place point on `superlongword`, and `M-x er/expand-region`. This shows the section with a code block!

```org
* Unrelated section without code...
* Unrelated section with code
... that contains some code:

#+BEGIN_SRC python
  print("Just some code!")
#+END_SRC

This section is expanded when expand-region is used.
* Section where expand-region is used
Let's say we have a superlongword. I want to select superlongword with
expand-region. When I =M-x expand-region= to select superlongword, the unrelated
section 
```

# Suggestion

I don't know who's to blame for this. Perhaps try to reproduce the behavior?

# System

- Spacemacs v.0.200.13 with org layer
- Expand-region from `~/.emacs.d/elpa/expand-region-20171217.2125`
- Ask if you need more information!

**Comment by ybyygu** [comment]
I can reproduce this annoying behavior. How to disable this feature?

**Comment by niklasfasching** [comment]
No time to look deeper into this right now but maybe this helps someone else out:
The problem seems to appear when `mark-page` is executed from `er/try-expand-list` via `er/expand-region-1`. 
Wrapping the loop through `er/try-expand-list` in `org-save-outline-visibility` in addition to `save-mark-and-excursion` fixes the problem for me - but that's not a clean solution.

**Comment by gahag** [comment]
@kevinjfoley does your commit fix this?  If so, could you open a pull request please? This is annoying me profoundly.

**Comment by magnars** [comment]
No-one has suggested a better solution, so let's get this in. 👍

**Comment by ytoku** [comment]
(require 'org) spoils autoload of org-mode.
How about using org-save-outline-visibility only when 'org is loaded?

```diff
diff --git a/expand-region-core.el b/expand-region-core.el
index b958d20..9a5b3a4 100644
--- a/expand-region-core.el
+++ b/expand-region-core.el
@@ -28,7 +28,6 @@
 
 (eval-when-compile (require 'cl))
 (require 'expand-region-custom)
-(require 'org)
 (declare-function er/expand-region "expand-region")
 
 (defvar er/history '()
@@ -66,6 +65,11 @@
     (defmacro save-mark-and-excursion (&rest body)
       `(save-excursion ,@body))))
 
+(defmacro er--org-save-outline-visibility (use-markers &rest body)
+  `(if (fboundp 'org-save-outline-visibility)
+       (eval '(org-save-outline-visibility ,use-markers . ,body))
+     . ,body))
+
 (defun er--expand-region-1 ()
   "Increase selected region by semantic units.
 Basically it runs all the mark-functions in `er/try-expand-list'
@@ -97,7 +101,7 @@ moving point or mark as little as possible."
       (setq start (point)))
 
     (while try-list
-      (org-save-outline-visibility t
+      (er--org-save-outline-visibility t
        (save-mark-and-excursion
          (ignore-errors
            (funcall (car try-list))
```

**Comment by magnars** [comment]
This seems to me like such a dirty hack tho.

The last two fixes I tried doing here both ended up not being good enough
for reasons I don’t care about (defcustom and autoload). I think it’s time
for me to stop trying to fix other people’s problems. I haven’t spent
nearly enough time on it to keep up with all the external needs. This
package works for me as is. Fork and hack it any way you want.

ons. 20. mar. 2019 kl. 17:00 skrev ytoku <notifications@github.com>:
